components/script/dom/htmlimageelement.rs

```diff
@@ -435,14 +435,24 @@ impl VirtualMethods for HTMLImageElement {
 }
 
 fn image_dimension_setter(element: &Element, attr: Atom, value: u32) {
+    use app_units::AU_PER_PX;
+    use std::i32;
+
     // This setter is a bit weird: the IDL type is unsigned long, but it's parsed as
     // a dimension for rendering.
     let value = if value > UNSIGNED_LONG_MAX {
         0
     } else {
         value
     };
-    let dim = LengthOrPercentageOrAuto::Length(Au::from_px(value as i32));
+
+    let pixel_value = if value > (i32::MAX / AU_PER_PX) as u32 {
```

Comment:
Yeah, I'll open an issue for this in app_units, and mention it from here.

I'd still like to land this before that is resolved, since that's potentially a big change, and I think the benefits of this are worth it.